packages/flutter/lib/src/material/search.dart

```diff
@@ -453,6 +453,7 @@ class _SearchPageState<T> extends State<_SearchPage<T>> {
             decoration: InputDecoration(
               border: InputBorder.none,
               hintText: searchFieldLabel,
+              hintStyle: theme.inputDecorationTheme.hintStyle,
```

Comment:
Sorry @guojiex, after thinking about this a little more, it seems like there exists a bug somewhere deeper. Inside `text_filed.dart` in `_getEffectiveDecoration` it should apply the theme values to the decoration and so the hintStyle should be passed along that way.

Could you please take a look as to why this isn't happening and see if there is fix that can be applied more broadly to TextFields? Thanks!

Reply:
When I run the unit test:
_getEffectiveDecoration() is called when render, and it can correctly show the hintstyle.
But the TextField.decoration is final and init only once, it will not show the hintstyle if not given(if the line in this pull request is not added).

But when I run a real app in emulator. At the time _getEffectiveDecoration is called, the hintstyle is always null, or TextStyle(inherit: true, color: Color(0xffffffff)), which is not overridden by appBarTheme function.

Reply:
@rami-a Sorry if I'm misunderstanding the discussion but to me it seems as the `ThemeData` (from `appbarTheme`) doesn't wrap the AppBar, rather the values are sort of 'cherry picked' from it in `search.dart`, when it comes to getting the `themeData.inputDecorationTheme` in `_getEffectiveDecoration`, the theme values are just those of the application. So this isn't a Text Field issue, rather one an issue with the search delegate?

Reply:
I agree with @NBTX.  `appBarTheme` provided in the SearchDelegate is not set as the theme for the context.  Instead, a few of its properties are "cherry picked" and set on the AppBar and the TextField.  When TextField grabs values from the theme, it is looking at the overall app theme and not SearchDelegate's `appBarTheme`.

If you set `hintColor` in the app theme then it will be respected by the Search TextField.  If you set it in appBarTheme it will be ignored (without this PR).

Overall it seems like this pattern is confusing and there is probably a better way to handle this sort of theming, but for now, I think this PR looks like a good improvement. @rami-a What do you think about moving forward with merging this?

Reply:
Yeah after looking at it more, I think this makes sense for now.